ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -120,38 +122,65 @@ class Endpoints(
                 )
               )
           }
-        } yield response
+        } yield transformedResponse
       }
     }
 
   private def logJsonRequestAndResult(implicit
       lc: LoggingContextOf[InstanceUUID with RequestID]
   ): Directive0 = {
-    def logWithHttpMessageBodyIfAvailable(
-        httpMessage: HttpMessage,
+    def logWithHttpMessageBodyIfAvailable[A <: HttpMessage](
+        httpMessage: A,
         msg: String,
-        bodyKind: String,
-    ): Future[Unit] =
+        kind: String,
+    ): HttpMessage =
       if (
         httpMessage
           .header[`Content-Type`]
           .map(_.contentType)
           .contains(ContentTypes.`application/json`)
-      )
-        httpMessage
-          .entity()
-          .toStrict(maxTimeToCollectRequest)
-          .map(it =>
+      ) {
+        httpMessage.entity.contentLengthOption match {
+          // Limit logging of bodies to content with size of less than 10 KiB
+          case Some(length) if length < 1024 * 10 =>
+            import akka.stream.scaladsl._
+            logger.debug("Calling transformEntityDataBytes")
+            httpMessage
+              .transformEntityDataBytes(
+                Flow.fromFunction { it =>
+                  try {
+                    withEnrichedLoggingContext(
+                      LoggingContextOf.label[RequestEntity],
+                      s"${kind}_body" -> it.utf8String.parseJson,
+                    )
+                      .run { implicit lc => logger.info(msg) }
+                  } catch {
+                    case ex: Exception =>
+                      logger.error("Failed to log message body: ", ex)
+                  }
+                  it
+                }
+              )
+          case other =>
+            val ctxContent = other
+              .map(length => s"OMITTED BECAUSE BODY SIZE OF $length IS TOO BIG FOR LOGGING")
+              .getOrElse {
+                if (httpMessage.entity.isChunked())
+                  "OMITTED BECAUSE REQUEST BODY IS CHUNKED & OVERALL SIZE IS UNKNOWN"
+                else
+                  "OMITTED BECAUSE REQUEST BODY SIZE IS UNKNOWN"
```

Comment:
This was probably good for debugging, but this should be capitalized normally going forward.

Reply:
OK :)